### PR TITLE
added warmup strategy

### LINKED ISSUE #5300: ebs br: support multiple warmup solution for tikv data volumes
## Feature Request

**Is your feature request related to a problem? Please describe:**

Hybrid warmup can bring jitter at restored cluster since not all cold blocks are scanned. 

**Describe the feature you'd like:**

We need to provide the additional option to do FIO / FSR based warmup to tikv data volumes

**Describe alternatives you've considered:**

**Teachability, Documentation, Adoption, Migration Strategy:**

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve?

For now, `warmup_steps` support some arguments that allow user to define where to be warmed up by fio(`--block`) or reading files(`--fs`), but there isn't an interface in `tidb-operator`.

Close #5300 

### What is changed and how does it work?
This PR added a new field `RestoreWarmupStrategy` to the restore CRD.
So the users are able to specify how to warmup the cluster.

### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [ ] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note
Added a new field `warmupStrategy ` to `Restore.Spec`, so user can choose how to warm up the cluster restored by EBS snapshot.
```

### CODE REVIEW

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 1]
```diff
@@ -2510,6 +2513,17 @@ const (
...
 
+type RestoreWarmupStrategy string
+
+const (
+	// RestoreWarmupStrategyAllByBlock warms up all data block by block. (use fio)
+	RestoreWarmupStrategyAllByBlock RestoreWarmupStrategy = "all-by-block"
+	// RestoreWarmupStrategyDataByFsOtherByBlock warms up data volume by read sst files one by one, other (e.g. WAL or Raft) will be warmed up via fio.
+	RestoreWarmupStrategyDataByFsOtherByBlock RestoreWarmupStrategy = "data-by-fs-other-by-block"
+	// RestoreWarmupStrategyDataByFSROtherByBlock warms up data volume by enabling Fast Snapshot Restore, other (e.g. WAL or Raft) will be warmed up via fio.
+	RestoreWarmupStrategyDataByFSROtherByBlock RestoreWarmupStrategy = "data-by-fsr-other-by-block"
```
Comment (BornChanger) [inline pkg/apis/pingcap/v1alpha1/types.go:2524]:
```suggestion
	RestoreWarmupStrategyAllByBlock RestoreWarmupStrategy = "fio"
	// RestoreWarmupStrategyDataByFsOtherByBlock warms up data volume by read sst files one by one, other (e.g. WAL or Raft) will be warmed up via fio.
	RestoreWarmupStrategyDataByFsOtherByBlock RestoreWarmupStrategy = "hybrid"
	// RestoreWarmupStrategyDataByFSROtherByBlock warms up data volume by enabling Fast Snapshot Restore, other (e.g. WAL or Raft) will be warmed up via fio.
	RestoreWarmupStrategyDataByFSROtherByBlock RestoreWarmupStrategy = "fsr"
```

**Review by BornChanger — APPROVED** [review]

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 2]
```diff
@@ -2479,6 +2479,9 @@ type RestoreSpec struct {
 	// WarmupImage represents using what image to initialize TiKV volumes
 	// +optional
 	WarmupImage string `json:"warmupImage,omitempty"`
+	// WarmupStrategy
+	// +kubebuilder:default=hybird
+	WarmupStrategy RestoreWarmupStrategy `json:"warmupStrategy,omitempty"`
```
Comment (BornChanger) [inline pkg/apis/pingcap/v1alpha1/types.go:2484]: We also need to change pkg/apis/federation/pingcap/v1alpha1/types.go

File: pkg/apis/pingcap/v1alpha1/types.go [hunk 3]
```diff
@@ -2510,6 +2513,18 @@ const (
...
 )
 
+// RestoreWarmupStrategy represents how to initialize TiKV volumes
+type RestoreWarmupStrategy string
+
+const (
+	// RestoreWarmupStrategyFio warms up all data block by block. (use fio)
+	RestoreWarmupStrategyFio RestoreWarmupStrategy = "fio"
+	// RestoreWarmupStrategyHybirx warms up data volume by read sst files one by one, other (e.g. WAL or Raft) will be warmed up via fio.
+	RestoreWarmupStrategyHybirx RestoreWarmupStrategy = "hybird"
```
Comment (WangLe1321) [inline pkg/apis/pingcap/v1alpha1/types.go:2523]:
```suggestion
	RestoreWarmupStrategyHybirx RestoreWarmupStrategy = "hybrid"
```

File: pkg/backup/restore/restore_manager.go [hunk 4]
```diff
@@ -1102,18 +1124,9 @@ func (rm *restoreManager) makeSyncWarmUpJob(r *v1alpha1.Restore, tc *v1alpha1.Ti
...
-	for _, volumeMount := range podVolumeMounts {
-		if volumeMount.MountPath == constants.TiKVDataVolumeMountPath {
-			continue
-		}
-		fioPaths = append(fioPaths, volumeMount.MountPath)
-	}
-	if len(fioPaths) > 0 {
-		args = append(args, "--block")
-		args = append(args, fioPaths...)
+	args, err := generateWarmUpArgs(r.Spec.WarmupStrategy, podVolumeMounts)
```
Comment (WangLe1321) [inline pkg/backup/restore/restore_manager.go:1127]: you need also modify args in async warmup

**Review by WangLe1321 — APPROVED** [review]

**Pull request merged**